SU-HSDO/suhumsci
# feat: add new image styles, create view modes, and expose them via wysiwyg

# READY FOR REVIEW

Note: Please do not merge until @pookmish gets a chance to look at this.

## Summary

The goal was to add new image styles to the WYSIWYG editor as requested by the H&S team:

![existing-wysiwyg-image-displays (1)](https://user-images.githubusercontent.com/1256329/75802689-200ee980-5d4b-11ea-8094-3684c8e7b6f9.png)

**New Image Styles**

- Large Square (480x480)
- Small Scaled (182px)
- Medium Scaled (352px)
- Large Scaled (480px)
- XL Scaled (610px)

**New View Modes**

- Small Square (stanford_image_small)
- XL Square (stanford_image_xl)
- Small Scaled (stanford_image_small_scaled)
- Medium Scaled (stanford_image_medium_scaled)
- Large Scaled (stanford_image_large_scaled)
- XL Scaled (stanford_image_xl_scaled)

**Updated View Modes**

- Large Square renamed to XL Square

## Need Review By (Date)
asap

## Urgency
medium

## Steps to Test
1. Pull down this branch
2. Import all the configuration for these image styles and view modes.
    - `lando drush @mysite.local config-import`
    - Warning: this will impact your local environment significantly:
3. Go to a basic page, create a Text Area Paragraph, insert each of the image styles and confirm that they work as expected (including with floats, etc)
4. Check for regressions.

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)
